core/src/test/scala/org/apache/spark/JobCancellationSuite.scala

```diff
@@ -153,6 +153,131 @@ class JobCancellationSuite extends SparkFunSuite with Matchers with BeforeAndAft
     assert(jobB.get() === 100)
   }
 
+  test("job tags") {
+    sc = new SparkContext("local[2]", "test")
+
+    // global ExecutionContext has only 2 threads in Apache Spark CI
+    // create own thread pool for four Futures used in this test
+    val numThreads = 4
+    val fpool = ThreadUtils.newForkJoinPool("job-tags-test-thread-pool", numThreads)
+    val executionContext = ExecutionContext.fromExecutorService(fpool)
```

Comment:
it seems to me this test was failing on CI, because the global implicit execution context there has only 2 threads...